enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+---
+title: ovn-kubernetes-ipsec
+authors:
+  - "@mdgray"
+reviewers:
+  - TBD
+  - "@dcbw"
+approvers:
+  - TBD
+creation-date: 2020-09-08
+last-updated: 2020-09-80
+status: provisional
+see-also:
+  - "/enhancements/20190919-OVN-Kubernetes.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# OVN Kubernetes IPsec Enablement
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OVN Kubernetes network plugin uses OVN (https://www.ovn.org) to instantiate
+virtual networks for Kubernetes. These virtual networks use Geneve, a network
+virtualization overlay encapsulation protocol, to tunnel traffic across the
+underlay network between Kubernetes Nodes.
+
+IPsec is a protocol suite that enables secure network communications between
+IP endpoints by providing the following services:
+- Confidentiality
+- Authentication
+- Data Integrity
+
+OVN tunnel traffic is transported by physical routers and switches. These
+physical devices could be untrusted (devices in public network) or might be
+compromised. Enabling IPsec encryption for this tunnel traffic can prevent the
+traffic data from being monitored and manipulated.
+
+This enhancement allows cluster operators to enable traffic to be encapsulated
+by the IPsec protocol between worker nodes.
```

Comment:
do you really mean just worker nodes?  Maybe "between all nodes" ?

Reply:
I wanted to specify that the focus of this was between worker nodes and not all nodes (i.e cluster mgmt traffic). As the pods will host cluster mgmt components, I guess this will be double encrypted. I don't think I will try to avoid this unless I can think of an easy way to do it but it is not the focus of this enhancement. Would it be clearer if I just say node everywhere instead of worker node?

Reply:
Yes, I have made this node/worker thing a little confusing! I will add a section at the top to try to clarify this and then change references throughout. It will state something like the following:

The scope of this work is to encrypt all traffic between pods on the cluster network (and not traffic between pods on the host network). As a side effect, this will also encrypt pod traffic that originates from pods on the host network.

Still confused? I am :S

I am going to resolve all the other comments on this theme and track this conversation here.